### PR TITLE
CA-368320: Update Json.NET from 10.0.2 to 13.0.1

### PR DESCRIPTION
**Pull request by danilo-delbusso**
About the changes:
- Had to update the `diff`s of `JsonConverter` and `JsonTextReader` because the files changed. However, the diffs are the same ones, just in different lines.
- I manually changed the assembly version in the `csproj` diff because the original repo does that at build time, else the `dll` would come out having version `11.0.0`
- I removed build target frameworks and only kept 4.5 and 4.8. This is easily amendable in the future if we want to target .Net Core
- You'll notice `dotnet restore "$SCRATCH_DIR\json48.net\Newtonsoft.Json\Newtonsoft.Json.csproj"` in the `build.ps1` script. This was apparently necessary because without it my machine wasn't generating some requested files
- I removed the following from `AssemblyInfo`'s diff because they were clashing with existing `.csproj` settings. The assembly comes out with the correct number so it doesn't seem to be an issue.
```diff
+[assembly: AssemblyVersion("10.0.2.1")]
+[assembly: AssemblyFileVersion("10.0.2.1")]
```
- We are building `Json.NET` twice, with the same settings. This is because I removed framework dependent diffs. We can get rid of it but I didn't want to touch the script too much, since for every other package we have a build for each .NET version.

I have tested the resulting `dll` using CHC's `master` as of the time of writing.

Played a bit and didn't find any issues.

A separate PR with changes to `CHC` will come once this one is merged

### CODE REVIEW

**Review by CitrixChris — DISMISSED** [review]

**Review by kc284 — CHANGES_REQUESTED** [review]

**Comment by kc284** [comment]
I've submitted two small comments in the first instance. Will add a couple more once we finish the discussion regarding the build script.

**Review by CitrixChris — APPROVED** [review]

File: build.ps1 [hunk 1]
```diff
@@ -131,30 +134,26 @@ Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-xmlrpc") -and !$_.Nam
...
 
-mkdirClean "$SCRATCH_DIR\json45.net"
-Expand-Archive -DestinationPath "$SCRATCH_DIR\json45.net" -Path "$REPO\Json.NET\Newtonsoft.Json-10.0.2.zip"
-Move-Item "$SCRATCH_DIR\json45.net\Newtonsoft.Json-10.0.2\Src\Newtonsoft.Json" "$SCRATCH_DIR\json45.net"
+Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-json-net")} |`
+  % { $_.FullName } | applyPatch -Path "$SCRATCH_DIR\json.net"
 
-Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-json-net") -and !$_.Name.Contains("dotnet48") } |`
-  % { $_.FullName } | applyPatch -Path "$SCRATCH_DIR\json45.net"
+& $msbuild $SWITCHES $RESTORE_SWITCHES -p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE $VS2019 $SIGN "$SCRATCH_DIR\json.net\Newtonsoft.Json\Newtonsoft.Json.csproj"
```
Comment (kc284) [inline build.ps1:152]: Suggestion: if you move the definition of `$RESTORE_NUGET_CONFIG_FILE` from line 144 to to line 70, you could compact this to:
`$RESTORE_SWITCHES= '/restore', '/p:RestoreNoCache=true', "/p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE"`

Comment (danilo-delbusso) [reply build.ps1:152]: I wanted to keep it close to where we unzip the source code and build it for better visibility in case of refactoring

**Comment by kc284** [comment]
This looks ok to me now, we just need one more commit to bump the pipeline version in the Jenkinsfile.

**Review by kc284 — APPROVED** [review]

**Comment by danilo-delbusso** [comment]
set it as draft because it depends on pipeline changes. Feel free to review anyway (it'll speed up merging)

**Comment by kc284** [comment]
Pipeline dependency done.

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**